examples/plugins/altemplate/src/suricata.rs

```diff
@@ -0,0 +1,547 @@
+// This file is kind of the include required by API
```

Comment:
this looks way to broad to me, even containing lots of things we define directly in rust. So why duplicate that? In general, we need to get to a point there this file is not needed at all.

Reply:
I see another project here, making Rust plugins a first class citizen.  All this "support" code should land in Suricata proper.

I have also needed to fill in some missing bits for an output plugin like this, just haven't brought them back into Suricata.  But we probably need some "suricata_plugin"  module or crate, that provides this support code to Rust, at least for the plugin specific stuff. Anything that's just filling in missing Suricata API should be in the main suricata rust package.

But it probably brings in more C into Rust, and maybe bindgen is worth another look instead of doing it by hand.

Reply:
> containing lots of things we define directly in rust.

Yes

> So why duplicate that?

Because my rust plugin needs to know the value of `IPPROTO_TCP` 
I would like to have some `#include <libsuricata.h>` but that does not work in rust

@jasonish do you have ideas ?